File: 프로그래머스/0/120838. 모스부호 （1）/모스부호 （1）.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(string letter) {
    map<string, char> morse = {
        {".-", 'a'},  {"-...", 'b'}, {"-.-.", 'c'}, {"-..", 'd'},
        {".", 'e'},   {"..-.", 'f'}, {"--.", 'g'},  {"....", 'h'},
        {"..", 'i'},  {".---", 'j'}, {"-.-", 'k'},  {".-..", 'l'},
        {"--", 'm'},  {"-.", 'n'},   {"---", 'o'},  {".--.", 'p'},
        {"--.-", 'q'},{".-.", 'r'},  {"...", 's'},  {"-", 't'},
        {"..-", 'u'}, {"...-", 'v'}, {".--", 'w'},  {"-..-", 'x'},
        {"-.--", 'y'},{"--..", 'z'}
    };
    
    string answer = "";
    string code = "";
    
    for(int i = 0; i < letter.size(); i++) {
        if(letter[i] == ' ') {
            answer += morse[code];
            code = "";
        } else {
            code += letter[i];
        }
    }
    answer += morse[code];
    return answer;
}

File: 프로그래머스/0/120838. 모스부호 （1）/모스부호 （1）.swift
import Foundation

func solution(_ letter:String) -> String {
    let morse: [String: String] = [
        ".-": "a",
        "-...": "b",
        "-.-.": "c",
        "-..": "d",
        ".": "e",
        "..-.": "f",
        "--.": "g",
        "....": "h",
        "..": "i",
        ".---": "j",
        "-.-": "k",
        ".-..": "l",
        "--": "m",
        "-.": "n",
        "---": "o",
        ".--.": "p",
        "--.-": "q",
        ".-.": "r",
        "...": "s",
        "-": "t",
        "..-": "u",
        "...-": "v",
        ".--": "w",
        "-..-": "x",
        "-.--": "y",
        "--..": "z"
    ]
    
    // let morseLetters = letter.split(separator: " ")
    // var result = ""
    // for morseCode in morseLetters {
    //     if let c = morse[String(morseCode)] {
    //         result += c
    //     }
    // }
    // return result
    
    return letter.components(separatedBy: " ").map { morse[$0] ?? "" }.reduce("", +) 
}